src/views/components/model.jsx

```diff
@@ -1,17 +1,65 @@
 import React from 'react';
-
+import ModelStore  from '../../store/model-store.js';
+import { searchModel } from '../../lib/api-client';
 export default class Model extends React.Component {
 
-    render() {
-        if (this.props.model === null) {
-            return <i>No model found</i>;
+    constructor(props){
+        super(props);
+        if (props.model === undefined)
+            this.state = {model: null};
+        else
+            this.state = {model: props.model};
+    }
+    componentDidMount(){
+        if(this.state.model === null) {
+            searchModel(this.props.params.address).then(() => {
+                this.setState({
+                    model: ModelStore.getModel()
+                });
+            });
         }
+    }
+
+    render() {
+        if(this.state.model === null)
+            return <div>Data is loading...</div>;
```

Comment:
Because eventually it was talked about having a loading bar instead of just text. And because I didn't care to check if they were both the same.